File: lwaf_3D/IPositioned.java
package lwaf_3D;

import lwaf_core.vec3;

public interface IPositioned<R> extends ITranslated<R> {
    default R setPosition(vec3 position) {
        return setTranslation(position);
    }

    default R moveBy(vec3 movement) {
        return translateBy(movement);
    }

    default R setPosition(float x, float y, float z) {
        return setTranslation(x, y, z);
    }

    default R moveBy(float x, float y, float z) {
        return translateBy(x, y, z);
    }
}

File: lwaf_3D/IPositioned.kt
package lwaf_3D

import lwaf_core.vec3

interface IPositioned<R> : ITranslated<R> {
    fun setPosition(position: vec3): R {
        return setTranslation(position)
    }

    fun moveBy(movement: vec3): R {
        return translateBy(movement)
    }

    fun setPosition(x: Float, y: Float, z: Float): R {
        return setTranslation(x, y, z)
    }

    fun moveBy(x: Float, y: Float, z: Float): R {
        return translateBy(x, y, z)
    }
}
